src/rotation.zig

```diff
@@ -321,6 +321,190 @@ pub const RotationModes = struct {
+				);
+				currentData.setConnection(neighbor.?, targetVal);
+
+				const result: u16 = currentData.enabledConnections;
+				if(result == currentBlock.data) return false;
+
+				currentBlock.data = result;
+				return true;
+			}
+			return false;
+		}
+
+		pub fn updateData(block: *Block, neighbor: Neighbor, neighborBlock: Block) bool {
+			const blockBaseModel = blocks.meshes.modelIndexStart(block.*);
+			const neighborBaseModel = blocks.meshes.modelIndexStart(neighborBlock);
+			var currentData = BranchData.init(block.data);
+
+			// Handle joining with other branches. While placed, branches extend in a
+			// opposite direction than they were placed from, effectively connecting
+			// to the block they were placed at.
+			if(blockBaseModel == neighborBaseModel) {
+				const neighborData = BranchData.init(neighborBlock.data);
+				currentData.setConnection(neighbor, neighborData.isConnected(neighbor.reverse()));
+			} else if(!neighborBlock.solid()) {
+				currentData.setConnection(neighbor, false);
+			}
+
+			const result: u16 = currentData.enabledConnections;
+			if(result == block.data) return false;
+
+			block.data = result;
+			return true;
+		}
+
+		fn closestRay(
+			block: Block,
+			relativePlayerPos: Vec3f,
+			playerDir: Vec3f
+		) ?u16 {
+			var resultIntersection: ?RayIntersectionResult = null;
+			var resultBitMask: ?u16 = null;
+			// Check intersection with central part of the branch block.
+			{
+				const modelIndex = blocks.meshes.modelIndexStart(block);
+				if(RotationMode.DefaultFunctions.rayModelIntersection(modelIndex, relativePlayerPos, playerDir)) |intersection| {
+					// We can't return here, as any intersection that accidentally passes
+					// through the central part would delete the whole block, even if player
+					// was aiming at the connection.
+					resultIntersection = intersection;
+					resultBitMask = 0;
+				}
+			}
+			// In case we did not intersect with the central part, check all the side
+			// connections.
+			for(Neighbor.iterable) |direction| {
+				const directionBitMask = Neighbor.bitMask(direction);
+
+				if((block.data & directionBitMask) != 0) {
+					const modelIndex = blocks.meshes.modelIndexStart(block) + directionBitMask;
+					if(RotationMode.DefaultFunctions.rayModelIntersection(modelIndex, relativePlayerPos, playerDir)) |intersection| {
+						// If we did have intersection with center part of the branch we must check
+						// if intersection we have right now is better than the one we had before.
+						// If we were to not do that, even accidental intersection with the connection
+						// would delete the connection even though player was aiming at the center.
+						if(resultIntersection == null or @abs(resultIntersection.?.distance) > @abs(intersection.distance)) {
+							return direction.bitMask();
+						}
+						// Since branch can have multiple connections, we must continue iteration
+						// in case connection with different intersection yields better results.
```

Comment:
That should be obvious.

Reply:
![iuy9fxt300811](https://github.com/user-attachments/assets/46ea3f4c-2fdb-4735-a742-b028b19f0b47)